doc/sphinx/grammar.rst

```diff
@@ -0,0 +1,2247 @@
+   hint : Resolve `reference_or_constr_list` `hint_info`
+        : Resolve -> `global_list` `natural_opt2`
+        : Resolve <- `global_list` `natural_opt2`
+        : Immediate `reference_or_constr_list`
+        : Variables Transparent
+        : Variables Opaque
+        : Constants Transparent
+        : Constants Opaque
+        : Transparent `global_list`
+        : Opaque `global_list`
+        : Mode `global` `mode`
+        : Unfold `global_list`
+        : Constructors `global_list`
+        : Extern `natural` `constr_pattern_opt` => `tactic`
+   reference_or_constr_list : `reference_or_constr_list` `reference_or_constr`
+                            : `reference_or_constr`
+   natural_opt2 : `natural`
+                : `empty`
+   constr_pattern_opt : `constr_pattern`
+                      : `empty`
+   constr_body : := `lconstr`
+               : : `lconstr` := `lconstr`
+   mode : `plus_list`
+   plus_list : `plus_list` `plus_alt`
+             : `plus_alt`
+   plus_alt : +
+            : !
+            : -
+   vernac_toplevel : Drop .
+                   : Quit .
+                   : Backtrack `natural` `natural` `natural` .
+                   : `vernac`
+   orient : ->
+          : <-
+          : `empty`
+   occurrences : `integer_list`
+               : `var`
+   integer_list : `integer_list` `integer`
+                : `integer`
+   glob : `constr`
+   lglob : `lconstr`
+   casted_constr : `constr`
+   hloc : `empty`
+        : in |- *
+        : in `ident`
+        : in ( Type of `ident` )
+        : in ( Value of `ident` )
+        : in ( type of `ident` )
+        : in ( value of `ident` )
+   rename : `ident` into `ident`
+   by_arg_tac : by `tactic_expr3`
+              : `empty`
+   in_clause : `in_clause`
+             : * `occs`
+             : * |- `concl_occ`
+             : `hypident_occ_list_opt` |- `concl_occ`
+             : `hypident_occ_list_opt`
+   hypident_occ_list_opt : `hypident_occ_list_opt` , `hypident_occ`
+                         : `empty`
+   test_lpar_id_colon : `empty`
+   withtac : with `tactic`
+           : `empty`
+   closed_binder : ( `name` `name_list` : `lconstr` )
+                 : ( `name` : `lconstr` )
+                 : ( `name` := `lconstr` )
+                 : ( `name` : `lconstr` := `lconstr` )
+                 : { `name` }
+                 : { `name` `name_list` : `lconstr` }
+                 : { `name` : `lconstr` }
+                 : { `name` `name_list` }
+                 : `( `typeclass_constraint_list` )
+                 : `{ `typeclass_constraint_list` }
+                 : ' `pattern0`
+                 : `of_alt` `operconstr99`
+                 : ( _ : `lconstr` | `lconstr` )
+   name_list : `name_list` `name`
+             : `name`
+   typeclass_constraint_list : `typeclass_constraint_list` , `typeclass_constraint`
+                             : `typeclass_constraint`
+   of_alt : of
+          : &
+   simple_tactic : reflexivity
```

Comment:
The `simple_tactic` token is the other typical example of what we don't want to have documented in the grammar but for which we prefer to use the `.. tac::` and `.. tacv::` directives. However, this tool would be useful to ensure that all supported tactic syntax are documented.

BTW, we should probably rename `simple_tactic` into `tactic`.

Reply:
My thought was to include `simple_tactic` and `command` in productionlist form only in the appendix (for completeness), using `tacn` etc. in the main part of the manual.  I'm open minded.

Reply:
I see, but I am still unsure of the usefulness of having an appendix with the full grammar. Maybe we had this discussion already though.